dashboard/modules/job/job_manager.py

```diff
@@ -379,8 +384,21 @@ async def run(
                 if sys.platform == "win32" and self._win32_job_object:
                     win32job.TerminateJobObject(self._win32_job_object, -1)
                 elif sys.platform != "win32":
-                    # TODO (jiaodong): Improve this with SIGTERM then SIGKILL
-                    child_process.kill()
+                    stop_signal = os.environ.get("RAY_JOB_STOP_SIGNAL", "SIGTERM")
+                    if stop_signal not in ["SIGINT", "SIGTERM"]:
```

Comment:
Should we limit the set of allowed signals, and if so what should they be?

Reply:
Why do we need to choose the signal type? Why not always use SIGINT?